viewing_party/party.py

```diff
@@ -0,0 +1,287 @@
+        for item in genre_list:
+            genre_dict[item] = 0
+        # Go through each genre in genre_list, and count how many times it's in there
+        # Add this count to the corresponding dictionary key in genre_dict
+        for item in genre_list:
+            if item in genre_dict.keys():
+                genre_dict[item] += 1
+
+        # Store the most_watched genre in a variable
+        # Use the max function to pass two arguments
+        # The first argument is the iterable (dictionary) to check
+        # The second argument accesses the key in genre_dict that has the highest value
+        # The .get function reference accesses the value of a particular key
+        most_watched = max(genre_dict, key=genre_dict.get)
+
+        print(most_watched)
+        return most_watched
+
+def get_unique_watched(user_data):
+    """
+    input: A dictionary of user_data
+    output: A list of movies as dictionaries, which were watched by the user but not their friends
+    """
+
+    user_set = set()
+    friend_set = set()
+
+    # Collect all of the movie titles watched by original user_data
+    # Access with user_data[0] <= loop through this list of dictionaries
+    # Grab the values at each title key and put into a set_1
+    for movie in user_data["watched"]:
+        user_set.add(movie["title"])
+
+    # this is a list - loop through this list of dictionaries
+    # in the dictionaries, access ["watched"], which is another list of dictionaries
+    # loop through the watched list to grab the ["title"] key and put that into a second set
+    for friend_data in user_data["friends"]:
+        for movie in friend_data["watched"]:
+            friend_set.add(movie["title"])
+    
+    # Compare both sets, getting the difference between set_1 and set_2
+    # Turn the resulting set into a list
+    # Return that list back to the function
+    unique_watched = list(user_set - friend_set)
+
+    new_list = []
+    # Create a dictionary for each title
+    # append that dictionary to unique watched with the key "title"
+    for movie in unique_watched:
+        new_list.append({"title": movie})
+
+    unique_watched = new_list
+    
+
+    return unique_watched
+
+def get_friends_unique_watched(user_data):
+    """
+    input: A dictionary of user_data
+    output: A list of movies that a user's friends have watched, but the user has not
+    """
+
+    user_set = set()
+    friend_set = set()
+
+    # Collect all of the movie titles watched by original user_data
+    # Access with user_data[0] <= loop through this list of dictionaries
+    # Grab the values at each title key and put into a set_1
+    for movie in user_data["watched"]:
+        user_set.add(movie["title"])
+    # Then access the second element in user_data[1]
+    for friend_data in user_data["friends"]:
+        for movie in friend_data["watched"]:
+            friend_set.add(movie["title"])
```

Comment:
Note that this code is the same as the code on lines 167-169. Consider making a helper function that returns `friends_set` to DRY up your code.